packages/app/src/client/views/ring-health/RingHealth.test.tsx

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright 2020 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import RingHealth from "./RingHealth";
+import Services from "client/services";
+import light from "client/themes/light";
+import ThemeProvider from "client/themes/Provider";
+import { StoreProvider } from "state/provider";
+import nock from "nock";
+import { MemoryRouter } from "react-router";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+import { render } from "@testing-library/react";
+
+jest.useFakeTimers();
+
+beforeEach(() => {
+  // @ts-ignore
+  nock.cleanAll();
+});
+
+const createMockShards = () => {
+  function getRandomInt() {
+    return Math.floor(Math.random() * 1000);
+  }
+  return new Array(5).fill(true).map(() => {
```

Comment:
It's mock data - creating 5 shards to be used as a mock request response - could be any other number as well.

Reply:
Ok then please leave a comment in the function saying something like: returns 5 shards with random ids used as mock request response

Reply:
I frankly believe if this is not obvious in the code, than the code needs to be improved. I'd suggest to rather work on making this more expressive.

Reply:
It might be obvious for people working on UI but for me, you're making have to scroll around to check how you're using this function. A comment explaining you need X amount of random data for the request-response would have been sufficient.

Reply:
> It might be obvious for people working on UI

I think the root issue was that it wasn't, and I wouldn't find that comment sufficient to make things clear enough (though that might be just me). So I did both, [added your suggested comment](https://github.com/opstrace/opstrace/pull/898/commits/cfd51fb6074273728ea162d48cb60cecfd188799#diff-56c2d53585831eec75922f289fd8883bb725f3f82c751cff8aad845ee5baef85R3) and tried to make things a bit clearer code wise.

Reply:
Thank you!